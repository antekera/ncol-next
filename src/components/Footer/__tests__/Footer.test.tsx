import React from 'react'

import { render } from '@testing-library/react'

import { Footer } from '..'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn()
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null)
    }
  }
}))

describe('Footer', () => {
  test('should match snapshots', () => {
    const { container } = render(<Footer />)
    expect(container).toMatchSnapshot()
  })
})
