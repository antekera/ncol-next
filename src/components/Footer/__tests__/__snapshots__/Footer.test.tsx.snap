// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshots 1`] = `
<div>
  <footer
    class="footer bg-dark-blue relative text-sm text-slate-300"
  >
    <button
      class="link-go-top bg-primary absolute -top-3 right-6 h-9 w-9 cursor-pointer rounded-sm border-none text-white duration-150 ease-in hover:-top-4"
      data-testid="button-go-top"
    >
      <svg
        class="lucide lucide-chevron-up"
        fill="none"
        height="36"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="36"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m18 15-6-6-6 6"
        />
      </svg>
    </button>
    <div
      class="bg-dark-blue text-xs"
    >
      <div
        class="container mx-auto px-6 sm:px-7 pt-12 pb-8"
      >
        <div
          class="flex flex-col md:flex-row"
        >
          <div
            class="col max-w-lg md:max-w-full md:basis-2/5 lg:basis-3/6 lg:pr-40"
          >
            <a
              class="link-logo"
              href="/"
            >
              <svg
                height="26"
                viewBox="0 0 409 72"
                width="140"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  id="Layer_1"
                >
                  <path
                    class="cls-2"
                    d="m0,18.05l13,0l0,6.18l0.2,0c2.8,-5.2 7.6,-7.57 13.39,-7.57c8.53,0 13.44,4.51 13.44,15l0,39.17l-13.53,0l0,-36.4c0,-5.39 -1.77,-7.36 -5.79,-7.36c-4.81,0 -7.16,2.94 -7.16,8.73l0,35l-13.55,0l0,-52.75z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-2"
                    d="m67.12,16.66c16.19,-0.29 20.8,9.32 20.8,27.77c0,18.05 -5.39,27.76 -20.8,27.76c-16.19,0.31 -20.83,-9.31 -20.83,-27.75c0,-18.05 5.42,-27.78 20.83,-27.78zm0,46.31c5.49,0 7.26,-5.3 7.26,-18.54s-1.76,-18.54 -7.26,-18.54c-6.67,0 -7.26,7.75 -7.26,18.54s0.58,18.55 7.26,18.55l0,-0.01z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-2"
                    d="m96.14,2.94l13.54,0l0,15.11l7.85,0l0,9.22l-7.85,0l0,28.55c0,4.22 0.89,5.78 4.81,5.78a28,28 0 0 0 3,-0.19l0,9.42a87,87 0 0 1 -8.83,0.59c-10.6,0 -12.56,-3.34 -12.56,-13.93l0,-30.22l-6.81,0l0,-9.22l6.87,0l-0.02,-15.11z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-2"
                    d="m122.75,0l13.54,0l0,11.57l-13.54,0l0,-11.57zm0,18.05l13.54,0l0,52.78l-13.54,0l0,-52.78z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-2"
                    d="m170.24,37.08c0,-5.1 -1,-10.59 -6.08,-10.59c-6.67,0 -7.75,5.89 -7.75,18.05c0,13 1.27,18.44 7.26,18.44c4.61,0 6.57,-3.73 6.57,-12.36l12.95,0c0,13.83 -5.69,21.58 -20.11,21.58c-13.54,0 -20.21,-6.57 -20.21,-27.76c0,-21.68 8.83,-27.78 21.42,-27.78s18.93,7.85 18.93,20.4l-12.98,0.02z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-2"
                    d="m189.77,0l13.52,0l0,11.57l-13.52,0l0,-11.57zm0,18.05l13.52,0l0,52.78l-13.52,0l0,-52.78z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-2"
                    d="m210.75,34.24l0,-1.28c0,-12.17 8.34,-16.3 18.54,-16.3c16.09,0 19.13,6.67 19.13,16.19l0,27.81c0,5 0.39,7.06 1.67,10.2l-13,0a20.92,20.92 0 0 1 -1.67,-5.5l-0.13,0c-3.14,5.5 -6.87,6.87 -13.44,6.87c-9.42,0 -12.75,-8 -12.75,-15.11c0,-10 4,-13.93 13.34,-16.48l7.65,-2.06c4,-1.08 5.4,-2.65 5.4,-6.18c0,-4 -1.67,-6.47 -6.28,-6.47c-4.22,0 -6.28,2.65 -6.28,6.87l0,1.47l-12.18,-0.03zm24.73,10.2a15.2,15.2 0 0 1 -5.79,2.65c-5.49,1.18 -7.65,3.83 -7.65,8.53c0,4 1.47,7.36 5.1,7.36s8.34,-2.26 8.34,-8.14l0,-10.4z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-2"
                    d="m267.97,53.56l0,1.77c0,4.61 2.45,7.65 7.07,7.65c4.22,0 6.87,-2.06 6.87,-6.28c0,-3.43 -2.26,-4.9 -4.91,-6l-9.52,-3.44c-7.46,-2.6 -11.29,-7.23 -11.29,-14.69c0,-8.63 5.59,-15.89 19.43,-15.89c12.26,0 18.05,5.49 18.05,15l0,2.16l-12.16,0c0,-5.4 -1.57,-7.95 -6,-7.95c-3.43,0 -6.38,1.86 -6.38,5.69c0,2.65 1.28,4.71 5.89,6.18l8.14,2.75c8.55,2.85 11.69,7.15 11.69,15.15c0,11.09 -8.24,16.58 -19.62,16.58c-15.3,0 -19.43,-6.57 -19.43,-16.78l0,-1.9l12.17,0z"
                    fill="#ffffff"
                  />
                  <path
                    class="cls-3"
                    d="m329.96,36.95c0,-5.1 -1,-10.59 -6.08,-10.59c-6.67,0 -7.75,5.89 -7.75,18.05c0,13 1.28,18.44 7.26,18.44c4.61,0 6.57,-3.73 6.57,-12.36l13,0c0,13.83 -5.69,21.58 -20.11,21.58c-13.54,0 -20.21,-6.57 -20.21,-27.76c0,-21.68 8.83,-27.77 21.39,-27.77s18.94,7.85 18.94,20.4l-13.01,0.01z"
                    fill="#f6921e"
                    transform="translate(-0.47, 0.13)"
                  />
                  <path
                    class="cls-3"
                    d="m368.03,16.53c16.19,-0.29 20.8,9.32 20.8,27.77c0,18.05 -5.4,27.76 -20.8,27.76c-16.19,0.29 -20.8,-9.32 -20.8,-27.76c0,-18.04 5.4,-27.77 20.8,-27.77zm0,46.31c5.5,0 7.26,-5.3 7.26,-18.54s-1.76,-18.54 -7.26,-18.54c-6.67,0.01 -7.27,7.77 -7.27,18.55s0.6,18.54 7.27,18.54l0,-0.01z"
                    fill="#fc8c29"
                    transform="translate(-0.47, 0.13)"
                  />
                  <path
                    class="cls-3"
                    d="m395.61,0.65l13.54,0l0,70l-13.54,0l0,-70z"
                    fill="#fc8c29"
                    transform="translate(-0.47, 0.13)"
                  />
                </g>
              </svg>
            </a>
            <h6
              class="pt-4 pr-4 leading-5"
            >
              El Diario Digital de la Costa Oriental del Lago, el Zulia y Venezuela. Noticias del Zulia, Cabimas, Maracaibo, Ciudad Ojeda, Caracas al d√≠a y las 24 horas.
            </h6>
            <div
              class="flex py-4"
            >
              <a
                class="link-social mr-6 hover:text-white"
                href="https://www.facebook.com/profile.php?id=61574619597032"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="w-4"
                  data-icon="facebook"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                    fill="currentColor"
                  />
                </svg>
              </a>
              <a
                class="link-social mr-6 hover:text-white"
                href="https://mobile.twitter.com/noticiasdelacol"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="w-4"
                  data-icon="x"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 310 380"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M178.57 127.15 290.27 0h-26.46l-97.03 110.38L89.34 0H0l117.13 166.93L0 300.25h26.46l102.4-116.59 81.8 116.59h89.34M36.01 19.54H76.66l187.13 262.13h-40.66"
                    fill="currentColor"
                  />
                </svg>
              </a>
              <a
                class="link-social mr-6 hover:text-white"
                href="https://www.instagram.com/noticiascol/"
                rel="noreferrer noopener"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="w-4"
                  data-icon="instagram"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </div>
          </div>
          <div
            class="col md:basis-1/5 md:pt-8 lg:basis-1/5"
          >
            <ul>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/costa-oriental"
                >
                  Noticias de la Costa Oriental
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/sucesos"
                >
                  Noticias de Sucesos
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/nacionales"
                >
                  Noticias Nacionales
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/economia"
                >
                  Noticias de Econom√≠a
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/politica"
                >
                  Noticias de Pol√≠tica
                </a>
              </li>
            </ul>
          </div>
          <div
            class="col md:basis-1/5 md:pt-8 lg:basis-1/5"
          >
            <ul>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/internacionales"
                >
                  Noticias Internacionales
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/salud"
                >
                  Noticias de Salud
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/curiosidades"
                >
                  Noticias de Curiosidades
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/tecnologia"
                >
                  Noticias de Tecnolog√≠a
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/categoria/deportes"
                >
                  Noticias de Deportes
                </a>
              </li>
            </ul>
          </div>
          <div
            class="col md:basis-1/5 md:pt-8 lg:basis-1/5"
          >
            <ul>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/contacto"
                >
                  Contacto
                </a>
              </li>
              <li
                class="list-none"
              >
                <a
                  class="link-footer inline-block pb-3 hover:text-white md:pb-2"
                  href="/publicidad"
                >
                  Publicidad
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bg-primary text-xs text-slate-300"
    >
      <div
        class="container mx-auto px-6 sm:px-7 py-2 text-center"
      >
        <span
          class="col"
        >
          2012 - 
          2025
           ¬© 
          Mas Multimedios C.A.
          J-40279329-4 
          <span
            class="hidden px-2 md:inline-block"
          >
            |
          </span>
          <br
            class="md:hidden"
          />
          <a
            class="link-bottom-bar hover:text-white"
            href="/terminos-y-condiciones"
          >
            T√©rminos y Condiciones
          </a>
          <span
            class="px-2"
          >
            |
          </span>
          <a
            class="link-bottom-bar hover:text-white"
            href="/privacidad"
          >
            Privacidad
          </a>
        </span>
      </div>
    </div>
  </footer>
</div>
`;
